universal-application-tool-0.0.1/app/services/applicant/ApplicantServiceImpl.java

```diff
@@ -410,6 +410,13 @@ private void stageEnumeratorUpdates(
             .map(Integer::valueOf)
             .collect(ImmutableList.toImmutableList());
     applicantData.deleteRepeatedEntities(enumeratorPath, deleteIndices);
+
+    // If there are no repeated entities at this point, we still need to save metadata for this
+    // question.
+    if (applicantData.readRepeatedEntities(enumeratorPath).isEmpty()) {
+      applicantData.maybeClearRepeatedEntities(enumeratorPath);
```

Comment:
(blocking) use the returned boolean to determine whether to write metadata or not here

Reply:
done